community/kernel/src/main/java/org/neo4j/kernel/internal/GraphDatabaseAPI.java

```diff
@@ -48,6 +52,43 @@
      */
     URL validateURLAccess( URL url ) throws URLAccessValidationError;
 
-    @Deprecated
     String getStoreDir();
+
+    /**
+     * Returns all relationship types currently in the underlying store.
+     * Relationship types are added to the underlying store the first time they
+     * are used in a successfully commited {@link Node#createRelationshipTo
+     * node.createRelationshipTo(...)}. Note that this method is guaranteed to
+     * return all known relationship types, but it does not guarantee that it
+     * won't return <i>more</i> than that (e.g. it can return "historic"
+     * relationship types that no longer have any relationships in the node
+     * space).
+     *
+     * @return all relationship types in the underlying store
+     */
+    ResourceIterable<RelationshipType> getAllRelationshipTypesInUseAndNot();
```

Comment:
@lutovich @MishaDemianenko @davidegrohmann guys, we're overloading the term "used" here. There are two types of used: (1) inUse records and (2) used as in there's at least 1 item in the database using this token.

(2) Doesn't even enter into what we're doing code-wise, although I think you're conflating (1) and (2) when discussing this.
The GDS#getAllXX methods returns (1), i.e. all token records that are inUse. This is fine and the only thing that we _can_ do, since we don't track how many are using the tokens. Nothing else makes sense imho.

The proposed introduced method `getAllRelationshipTypesInUseAndNot` would return all inUse + !inUse records (again (1)), right? I dunno why that would be necessary... are there tests using that perhaps. I think even having this method is odd, unless perhaps in a test class somewhere. What exists in GDS is good and correct as it is, even the javadoc is correct since it says

```
...it can return "historic" relationship types that no longer have any relationships in the node pace).
```

this is what it does and the only thing we can reasonably do.